contracts/protocols/mainnet/universeFinance/main.sol

```diff
@@ -19,43 +18,58 @@ abstract contract UniverseFinanceResolver is Helpers {
      * @return [token0Address, token1Address, vaultMaxToken0Amount, vaultMaxToken1Amount, maxSingleDepositFofToken0,
      maxSingleDepositFofToken1, totalToken0Amount, totalTotal1Amount, utilizationOfToken0, utilizationOfToken1]
      */
-    function getVaultDetail(address universeVault) external view returns(VaultData memory) {
+    function getVaultDetail(address universeVault) public view returns (VaultData memory) {
         return _vaultDetail(universeVault);
     }
 
+    /**
+     * @notice get universe vaults data
+     * @param universeVaults the Universe Vault Address array
+     * @return array of [token0Address, token1Address, vaultMaxToken0Amount, vaultMaxToken1Amount, 
+        maxSingleDepositFofToken0,maxSingleDepositFofToken1, totalToken0Amount, totalTotal1Amount, 
+        utilizationOfToken0, utilizationOfToken1]
+     */
+    function getVaultData(address[] memory universeVaults) external view returns (VaultData[] memory) {
```

Comment:
Need to add one more function like this where we are not sending any vaults and it's directly fetching it from `getAllVault()`.